mlir-tensorrt/python/bindings/Runtime/RuntimePyBind.cpp

```diff
@@ -950,22 +959,43 @@ PYBIND11_MODULE(_api, m) {
       .def(
           "execute_function",
           [](PyRuntimeSession &self, std::string name,
-             std::vector<py::object> inArgs, std::vector<py::object> outArgs,
-             std::optional<MTRT_Stream> stream) {
+             std::vector<py::object> inArgs,
+             std::optional<std::vector<py::object>> outArgs,
+             std::optional<MTRT_Stream> stream, PyRuntimeClient &client) {
```

Comment:
I have tried making `PyRuntimeClient` optional but for some reason I am getting argument mismatch error while calling the function.
```
TypeError: execute_function(): incompatible function arguments. The following argument types are supported:
    1. (self: mlir_tensorrt.runtime._mlir_libs._api.RuntimeSession, name: str, in_args: List[object], out_args: Optional[List[object]] = None, stream: Optional[MTRT_Stream] = None, client: Optional[MTRT_RuntimeClient] = None) -> List[object]

Invoked with: <mlir_tensorrt.runtime._mlir_libs._api.RuntimeSession object at 0x7f86f29f88f0>, 'main'; kwargs: in_args=[<mlir_tensorrt.runtime._mlir_libs._api.MemRefValue object at 0x7f86f29e9f70>], stream=<mlir_tensorrt.runtime._mlir_libs._api.Stream object at 0x7f86f2c682b0>, client=<mlir_tensorrt.runtime._mlir_libs._api.RuntimeClient object at 0x7f86f2c68270>
```

Reply:
did you add `= py::none()` as the default value of `py::arg("client")`

Reply:
Yes. I did.

Reply:
It looks like using `PyRuntimeClient*` works instead of using `std::optional<PyRuntimeClient>`.